sound/soc/sof/intel/hda-dsp.c

```diff
@@ -485,8 +494,16 @@ int hda_dsp_runtime_suspend(struct snd_sof_dev *sdev)
 int hda_dsp_suspend(struct snd_sof_dev *sdev)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct pci_dev *pci = to_pci_dev(sdev->dev);
 	int ret;
 
+	if (sdev->s0_suspend) {
+		/* suspending to S0ix */
+		enable_irq_wake(pci->irq);
+		pci_save_state(pci);
+		return 0;
+	}
```

Comment:
commit message :
ASoC: SOF: Intel: hda-dsp: Enable system wake up via IPC interrupt

 Enable system wake up via IPC interrupt from DSP when the system is suspending to the S0ix state

Reply:
It is not only enabling the waking up via IPC interrupt but also with totally different suspend/resume sequence, compared with the D3 case.

ASoC: SOF: Intel: hda-dsp: implement d0i3 suspend/resume

For D0i3 suspending, we will keep the HDA controller and the DSP what they are, and enable the system waking up via IPC interrupt triggered by the DSP firmware.

Reply:
@keyonjie I dont understand. The patch only deals with what do to in case of S0ix suspend/resume ie enable/disable pci irq wake when entering S0ix. So the commit message below makes far more sense. If you want to add that the HDA controller and DSP are not powered off in S0ix that should fine. But why talk about D0i3?

ASoC: SOF: Intel: hda-dsp: Enable system wake up via IPC interrupt

Enable system wake up via IPC interrupt from DSP when the system is suspending to the S0ix state.

Reply:
right, what we are doing here is S0ix specific, not d0i3 related, but yes, I am adding more stuff into it, not only IPC wake up inside it.

Reply:
How about using this?

ASoC: SOF: Intel: hda-dsp: implement suspend/resume for S0ix<->S0 transition

When the system is suspending to the S0ix state, we will enable the DMI L1 support, and enable waking up the systeam via IPC interrupt from the DSP.

Reply:
you need to explain why you keep track of state changes.

Reply:
> you need to explain why you keep track of state changes.

keeping track of L1SEN is needed fore restoring it on resuming, how should I word this in the commit message, @plbossart ?

Reply:
not sure what you did here, you removed the code for L1 but kept it in the commit message?